services/QuillLMS/app/services/concerns/open_ai/api.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module OpenAI
+  module Api
+    extend ActiveSupport::Concern
+
+    BASE_URI = 'https://api.openai.com/v1'
+    TIMEOUT = 5.minutes.to_i
+
+    included do
+      include HTTParty
+      base_uri BASE_URI
+
+      attr_accessor :response
+    end
+
+    # classes using this concern require these methods
+    def endpoint
+      raise NotImplementedError
+    end
+
+    def request_body
+      raise NotImplementedError
+    end
+
+    def cleaned_results
+      raise NotImplementedError
+    end
+
+    def run
+      return cleaned_results if response.present?
+
+      @response = post_request
+
+      cleaned_results
+    rescue *Evidence::HTTP_TIMEOUT_ERRORS
+      []
+    end
```

Comment:
This one I'm gonna leave since it's a straight copy of the version in the `Evidence` engine, and if we want to bring them back together later better if they're more identical.